### DISCUSSION TITLE
Calling refreshDatatable with livewire component causes duplicate rows / "Cannot read properties of null (reading 'fingerprint')"

### DISCUSSION THREAD

**Discussion by idealerror** [rappasoft/laravel-livewire-tables#1057] [Q&A]
I have a DataTable that has an "Actions" Column that calls a view which calls a Livewire component.  When I call refreshDatatable, it essentially duplicates the data in the table instead of removing it.  Is this caused by calling a livewire component within a view?

Is there a best practice for this?  I'd like to delete a row and refresh the table so the row is removed, but could not find a good way to do this.  

Everything works, except when calling `delete()` in the Livewire component, it refreshes the table and duplicates data.  The delete works though.  In the console I see this error:

```javascript
Cannot read properties of null (reading 'fingerprint')
```

Here's an overview of how this is implemented:

> DataTable
```php
Column::make('')
->label(
function ($row, Column $column) {
    return view('components.actions', ['request' => $row->uuid]);
})
->excludeFromColumnSelect(),
```

> components/actions.blade.php view
```php
<livewire:table-actions :uuid="$request" />
```

> table-actions.blade.php
```php
<a wire:click="delete()" href="#" onClick="return confirm('Are you sure you want to delete this request?') || event.stopImmediatePropagation()" class="btn-xs border-slate-200 hover:border-slate-300 text-rose-500">
        <svg class="w-4 h-4 fill-current shrink-0" viewBox="0 0 16 16">
            <path d="M5 7h2v6H5V7zm4 0h2v6H9V7zm3-6v2h4v2h-1v10c0 .6-.4 1-1 1H2c-.6 0-1-.4-1-1V5H0V3h4V1c0-.6.4-1 1-1h6c.6 0 1 .4 1 1zM6 2v1h4V2H6zm7 3H3v9h10V5z" />
        </svg>
        <span class="ml-2">Delete</span>
    </a>
```

> TableActions component
```php
public function delete()
    {
        $this->request->delete();
        $this->dispatchBrowserEvent('alert',
            ['type' => 'success',  'message' => 'Deleted!']);
        $this->emit('refreshDatatable');
    }
```

**Comment by lrljoe** [comment]
Better to use the shiny new [Component Column type](https://rappasoft.com/docs/laravel-livewire-tables/v2/columns/other-column-types#content-component-columns
) :

```php
ComponentColumn::make('E-mail', 'email')
    ->component('email')
    ->attributes(fn ($value, $row, Column $column) => [
        'type' => Str::endsWith($value, 'example.org') ? 'success' : 'danger',
        'dismissible' => true,
    ]),
```
